vendor/src/github.com/docker/libnetwork/portmapper/mapper.go

```diff
@@ -128,8 +128,10 @@ func (pm *PortMapper) MapRange(container net.Addr, hostIP net.IP, hostPortStart,
 	}
 
 	containerIP, containerPort := getIPAndPort(m.container)
-	if err := pm.forward(iptables.Append, m.proto, hostIP, allocatedHostPort, containerIP.String(), containerPort); err != nil {
-		return nil, err
+	if hostIP.To4() != nil {
+		if err := pm.forward(iptables.Append, m.proto, hostIP, allocatedHostPort, containerIP.String(), containerPort); err != nil {
+			return nil, err
+		}
```

Comment:
> Also a general question, why do you need the port translation for IPv6 ?

The idea is to have the ability to bind to ipv6 interface only. It's possible with ipv4 but not with ipv6.